### PR TITLE
8282789: Create a regression test for the JTree usecase of JDK-4618767

### PR DESCRIPTION
**Pull request by manukumarvs**
Create a regression test for the JTree usecase of [JDK-4618767](https://bugs.openjdk.java.net/browse/JDK-4618767)

Issue identified in [JDK-4618767](https://bugs.openjdk.java.net/browse/JDK-4618767):
Typing a letter while a JTree has focus now makes the selection jump to the first/next node/item whose text starts with that letter even though that letter is accompanied by modifier keys such as ALT or CTRL.

Fix:
Only enable JTree letter navigation when the user doesn't press any modifier keys such as ALT or CTRL.

Testing:
Java 1.4.0 -> Test Failed.
$ ./j2sdk1.4.0/bin/java JTreeOld
LookAndFeel: javax.swing.plaf.metal.MetalLookAndFeel
OS = windows nt (unknown)
java.lang.RuntimeException: Test Failed, expected colors, but got food
        at JTreeOld.doTest(JTreeOld.java:166)
        at JTreeOld.main(JTreeOld.java:32)
Exception in thread "main"

Java 1.4.1 -> Test Passed.
$ ./j2sdk1.4.1/bin/java JTreeOld
LookAndFeel: javax.swing.plaf.metal.MetalLookAndFeel
OS = windows nt (unknown)
Test passed javax.swing.plaf.metal.MetalLookAndFeel
LookAndFeel: com.sun.java.swing.plaf.motif.MotifLookAndFeel
OS = windows nt (unknown)
Test passed com.sun.java.swing.plaf.motif.MotifLookAndFeel
LookAndFeel: com.sun.java.swing.plaf.windows.WindowsLookAndFeel
OS = windows nt (unknown)
Test passed com.sun.java.swing.plaf.windows.WindowsLookAndFeel

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8282789](https://bugs.openjdk.java.net/browse/JDK-8282789): Create a regression test for the JTree usecase of JDK-4618767

### Reviewers
 * [Alexey Ivanov](https://openjdk.java.net/census#aivanov) (@aivanov-jdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7742/head:pull/7742` \
`$ git checkout pull/7742`

Update a local copy of the PR: \
`$ git checkout pull/7742` \
`$ git pull https://git.openjdk.java.net/jdk pull/7742/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7742`

View PR using the GUI difftool: \
`$ git pr show -t 7742`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7742.diff">https://git.openjdk.java.net/jdk/pull/7742.diff</a>

</details>

### CODE REVIEW

**Comment by manukumarvs** [comment]
@aivanov-jdk Can you please review this.

**Comment by manukumarvs** [comment]
Any volunteers?

File: test/jdk/javax/swing/JTree/4618767/JTreeSelectedElementTest.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        });
+        return nodeName.get();
+    }
+
+    private static Point getNodeLocation(int rowCount) throws Exception {
+        AtomicReference<Point> treeNodeLoc = new AtomicReference<>();
+        SwingUtilities.invokeAndWait(() -> {
+            final Point locationOnScreen = tree.getLocationOnScreen();
+            Rectangle rt = tree.getPathBounds(tree.getPathForRow(rowCount));
+            locationOnScreen.translate((int) (rt.getX() + rt.getWidth() / 2), (int) (rt.getY() + rt.getHeight() / 2));
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/JTree/4618767/JTreeSelectedElementTest.java:166]:
```suggestion
            locationOnScreen.translate(rt.x + rt.width / 2, rt.y + rt.height / 2);
```

**Review by aivanov-jdk — APPROVED** [review]

**Review by aivanov-jdk — APPROVED** [review]

**Pull request closed without merging**